glvis@4.2 build fails: fatal error: GL/glu.h: No such file or directory

### Steps to reproduce the issue

`glvis@4.2` build fails using:
* spack @ e4s-24.02 (2064934ed3c02edeab05bbd5719df6acf5743784 from `Fri Jan 26 13:56:16 2024 -0800`)
* Ubuntu 20.04, AMD64
* GCC 11.4.0

Spack environment: [spack.yaml.txt](https://github.com/spack/spack/files/14393765/spack.yaml.txt)

<details><summary>Concretization</summary>

<pre>
 -   io2djnj  glvis@4.2%gcc@11.4.0+fonts build_system=makefile screenshots=png arch=linux-ubuntu20.04-x86_64
[+]  vos6ois      ^fontconfig@2.14.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  kifzxsd          ^font-util@1.4.0%gcc@11.4.0 build_system=autotools fonts=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-ubuntu20.04-x86_64
[+]  rp4masv              ^autoconf@2.72%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  nemwy7w                  ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]  pmupyim                      ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ehu7kq2              ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  tyd5zkb              ^bdftopcf@1.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  onrxppe                  ^fontsproto@2.1.3%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  6pnv5or                  ^libxfont@1.5.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  f6k5p5y              ^mkfontdir@1.0.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  vr2tqou              ^mkfontscale@1.2.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  r6c3eph                  ^libfontenc@1.1.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  7kyxnhc              ^util-macros@1.19.3%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ciquhh6          ^gperf@3.1%gcc@11.4.0 build_system=autotools patches=3dd36db arch=linux-ubuntu20.04-x86_64
[+]  aop7bw3          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  aqsldim              ^libiconv@1.17%gcc@11.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  srvsu5c              ^xz@5.4.1%gcc@11.4.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  qrvfe44          ^pkgconf@1.9.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  uubpzwi          ^python@3.11.6%gcc@11.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]  tnrtanf              ^expat@2.5.0%gcc@11.4.0+libbsd build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  aasyfru                  ^libbsd@0.11.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  3g4esl5                      ^libmd@1.0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  psgictj              ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  2s4dz5h              ^gettext@0.22.4%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  3khlwqv                  ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-x86_64
[+]  5nhltww                      ^pigz@2.8%gcc@11.4.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]  xlag2og                      ^zstd@1.5.5%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  ampwpxw              ^libffi@3.4.4%gcc@11.4.0 build_system=autotools patches=070b1f3 arch=linux-ubuntu20.04-x86_64
[+]  ohf4eow              ^libxcrypt@4.4.35%gcc@11.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu20.04-x86_64
[+]  3wuihfw              ^ncurses@6.3%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  smzflud              ^openssl@3.1.3%gcc@11.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-x86_64
[+]  rjzhk7z                  ^ca-certificates-mozilla@2023-05-30%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  zklrfj4              ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
[+]  c6t6exn              ^sqlite@3.43.2%gcc@11.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  5qdbe7u          ^util-linux-uuid@2.38.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  pejvlxs      ^freetype@2.11.1%gcc@11.4.0+pic+shared build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  adpnfj3          ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  remvexl              ^diffutils@3.9%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  47cdzlk      ^gcc-runtime@11.4.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  gysfwmr      ^glew@2.2.0%gcc@11.4.0~ipo build_system=cmake build_type=Release generator=make gl=glx patches=7992e52 arch=linux-ubuntu20.04-x86_64
[+]  kwqn5nw          ^cmake@3.27.9%gcc@11.4.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  75is7we              ^curl@8.4.0%gcc@11.4.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu20.04-x86_64
[+]  ego5ibp                  ^nghttp2@1.57.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  geyfuv4          ^libx11@1.8.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  imf4hq4              ^inputproto@2.3.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  3wbcjrq              ^kbproto@1.0.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  plua4a5              ^libxcb@1.16%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  qtjzt4k                  ^libpthread-stubs@0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ve2upwm                  ^libxau@1.0.11%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  b3dyuuf                  ^libxdmcp@1.1.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  h47ztz2                  ^xcb-proto@1.16.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ofbgoud              ^perl@5.38.0%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu20.04-x86_64
[+]  3n37avu                  ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]  uhtwu62              ^xextproto@7.3.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  bib3iq2              ^xtrans@1.5.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  47qlg4l          ^xproto@7.0.31%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  jo5b6ou      ^glm@0.9.9.8%gcc@11.4.0~ipo build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
[+]  pvqt2h3      ^glx@1.4%gcc@11.4.0 build_system=bundle arch=linux-ubuntu20.04-x86_64
[+]  iibx3cf          ^mesa@23.2.1%gcc@11.4.0+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared arch=linux-ubuntu20.04-x86_64
[+]  l5dag3h              ^bison@3.8.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  x3fk5nc              ^flex@2.6.3%gcc@11.4.0+lex~nls build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  owbho3g                  ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]  bocxtol              ^glproto@1.4.17%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  gda2zh7              ^libunwind@1.6.2%gcc@11.4.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib build_system=autotools components=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  7amqdnv              ^libxt@1.3.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  gau3tzq                  ^libice@1.1.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  7inli27                  ^libsm@1.2.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  vy4zlhe              ^llvm@17.0.4%gcc@11.4.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime shlib_symbol_version=none targets=all version_suffix=none arch=linux-ubuntu20.04-x86_64
[+]  tybzbvk                  ^binutils@2.41%gcc@11.4.0~gas+gold~gprofng+headers~interwork+ld+libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  vawbq73                  ^hwloc@2.9.1%gcc@11.4.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  qrwjdjb                      ^libpciaccess@0.17%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  xlpsje4                          ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  uoken6t                  ^libedit@3.1-20210216%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  xrz5aow                  ^lua@5.3.6%gcc@11.4.0~pcfile+shared build_system=makefile fetcher=curl arch=linux-ubuntu20.04-x86_64
[+]  uu3rowm                      ^unzip@6.0%gcc@11.4.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]  2saqk73                  ^perl-data-dumper@2.173%gcc@11.4.0 build_system=perl arch=linux-ubuntu20.04-x86_64
[+]  jzchgxz                  ^swig@4.1.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  zopggb7                      ^pcre2@10.42%gcc@11.4.0~jit+multibyte build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  zz6vamr              ^meson@1.2.2%gcc@11.4.0 build_system=python_pip patches=0f0b1bd,ae59765 arch=linux-ubuntu20.04-x86_64
[+]  6plkpsr                  ^py-pip@23.1.2%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  yv7s32i                  ^py-setuptools@68.0.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  jifyxkm                  ^py-wheel@0.41.2%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  o5z7lfg              ^ninja@1.11.1%gcc@11.4.0+re2c build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  xkeius5                  ^re2c@2.2%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  janlpts              ^py-mako@1.2.4%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  xjabq4l                  ^py-markupsafe@2.1.3%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  rkjezet              ^xrandr@1.5.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  p33tip5                  ^libxrandr@1.5.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  kgzfr2c                      ^renderproto@0.11.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ob3zpv4                  ^libxrender@0.9.11%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  yojcvcf                  ^randrproto@1.5.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  ldw6zsv      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  z355xqx      ^libpng@1.6.39%gcc@11.4.0~ipo~pic build_system=cmake build_type=Release generator=make libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  5x6i3wn          ^zlib-ng@2.1.5%gcc@11.4.0+compat+opt build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  b7q3ztq      ^mfem@4.6.0%gcc@11.4.0~amgx~conduit~cuda~debug~examples~exceptions~fms~ginkgo~gnutls~gslib~hiop~lapack~libceed~libunwind+metis~miniapps~mpfr+mpi~netcdf~occa~openmp~petsc~pumi~raja~rocm~shared~slepc+static~strumpack~suite-sparse~sundials~superlu-dist~threadsafe~umpire+zlib build_system=generic patches=718f073 timer=auto arch=linux-ubuntu20.04-x86_64
[+]  tlpkuqe          ^hypre@2.30.0%gcc@11.4.0~caliper~complex~cuda~debug+fortran~gptune~int64~internal-superlu~magma~mixedint+mpi~openmp~rocm+shared~superlu-dist~sycl~umpire~unified-memory build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  lktjge3              ^openblas@0.3.26%gcc@11.4.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu20.04-x86_64
[+]  wkqolmo          ^metis@5.1.0%gcc@11.4.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu20.04-x86_64
[e]  qo3z6y6          ^mpich@4.1.2%gcc@11.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]  4kr3mg4      ^sdl2@2.26.5%gcc@11.4.0~ipo build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
[+]  wngrsjh          ^libxext@1.3.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  otzk6bz      ^xxd-standalone@8.2.1201%gcc@11.4.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
</pre></details>


Error:
```
$> spack -e . install
...
==> Installing glvis-4.2-io2djnjwww3nfp4v5jtx6uqx3obj6dkx [83/83]
==> No binary for glvis-4.2-io2djnjwww3nfp4v5jtx6uqx3obj6dkx found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/31/314fb04040cd0a8128d6dac62ba67d7067c2c097364e5747182ee8371049b42a..tar.gz
==> No patches needed for glvis
==> glvis: Executing phase: 'edit'
==> glvis: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'CC=/spack/lib/spack/env/gcc/gcc' 'PREFIX=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/glvis-4.2-io2djnjwww3nfp4v5jtx6uqx3obj6dkx/bin' 'MFEM_DIR=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/mfem-4.6.0-b7q3ztqv27llozzw73k6fiw4c2civosh' 'CONFIG_MK=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/mfem-4.6.0-b7q3ztqv27llozzw73k6fiw4c2civosh/share/mfem/config.mk' 'GLM_DIR=' 'SDL_DIR=' 'GLEW_DIR=' 'FREETYPE_DIR=' 'OPENGL_DIR=' 'GLVIS_USE_LIBPNG=YES' 'GLVIS_USE_LIBTIFF=NO' 'PNG_OPTS=-DGLVIS_USE_LIBPNG -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/libpng-1.6.39-z355xqx64mdsbdluyrzvik62zissuker/include' 'PNG_LIBS=-L/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/libpng-1.6.39-z355xqx64mdsbdluyrzvik62zissuker/lib -lpng16'

32 errors found in build log:
     20     /usr/local/mpich/install/mpich/bin/mpic++ -O3 -std=c++11 -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/mfem-4.6.0-b7q3ztqv27llozzw73k6fiw4c2civosh/include -I/spack/opt/spack/linux
            -ubuntu20.04-x86_64/gcc-11.4.0/hypre-2.30.0-tlpkuqe7cklavodwkaj3yclqzqfgqjsy/include -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/metis-5.1.0-wkqolmovas7we2lueutxa6ntyuik2gf7/inc
            lude -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/zlib-ng-2.1.5-5x6i3wn4nd3uvng3wggbb6n7jjqrqpv6/include -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.4 -DGLVIS_OGL3 -DGLVIS_USE_L
            OGO -DGLVIS_USE_LIBPNG -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/libpng-1.6.39-z355xqx64mdsbdluyrzvik62zissuker/include -o lib/vsdata.o -c lib/vsdata.cpp
     21     /usr/local/mpich/install/mpich/bin/mpic++ -O3 -std=c++11 -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/mfem-4.6.0-b7q3ztqv27llozzw73k6fiw4c2civosh/include -I/spack/opt/spack/linux
            -ubuntu20.04-x86_64/gcc-11.4.0/hypre-2.30.0-tlpkuqe7cklavodwkaj3yclqzqfgqjsy/include -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/metis-5.1.0-wkqolmovas7we2lueutxa6ntyuik2gf7/inc
            lude -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/zlib-ng-2.1.5-5x6i3wn4nd3uvng3wggbb6n7jjqrqpv6/include -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.4 -DGLVIS_OGL3 -DGLVIS_USE_L
            OGO -DGLVIS_USE_LIBPNG -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/libpng-1.6.39-z355xqx64mdsbdluyrzvik62zissuker/include -o lib/vssolution.o -c lib/vssolution.cpp
     22     /usr/local/mpich/install/mpich/bin/mpic++ -O3 -std=c++11 -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/mfem-4.6.0-b7q3ztqv27llozzw73k6fiw4c2civosh/include -I/spack/opt/spack/linux
            -ubuntu20.04-x86_64/gcc-11.4.0/hypre-2.30.0-tlpkuqe7cklavodwkaj3yclqzqfgqjsy/include -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/metis-5.1.0-wkqolmovas7we2lueutxa6ntyuik2gf7/inc
            lude -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/zlib-ng-2.1.5-5x6i3wn4nd3uvng3wggbb6n7jjqrqpv6/include -DGLVIS_MULTISAMPLE=4 -DGLVIS_MS_LINEWIDTH=1.4 -DGLVIS_OGL3 -DGLVIS_USE_L
            OGO -DGLVIS_USE_LIBPNG -I/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/libpng-1.6.39-z355xqx64mdsbdluyrzvik62zissuker/include -o lib/vssolution3d.o -c lib/vssolution3d.cpp
     23     In file included from lib/gl/platform_gl.hpp:15,
     24                      from lib/gl/renderer.hpp:20,
     25                      from lib/gl/renderer.cpp:12:
  >> 26     /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/glew-2.2.0-gysfwmrf7cotyfdggpuw6o7p462iyszv/include/GL/glew.h:1219:14: fatal error: GL/glu.h: No such file or directory
     27      1219 | #    include <GL/glu.h>
     28           |              ^~~~~~~~~~
     29     In file included from lib/gl/platform_gl.hpp:15,
     30                      from lib/gl/renderer.hpp:20,
     31                      from lib/gl/shader.hpp:15,
     32                      from lib/gl/shader.cpp:12:
  >> 33     /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/glew-2.2.0-gysfwmrf7cotyfdggpuw6o7p462iyszv/include/GL/glew.h:1219:14: fatal error: GL/glu.h: No such file or directory
     34      1219 | #    include <GL/glu.h>
     35           |              ^~~~~~~~~~
     36     compilation terminated.
     37     compilation terminated.
  >> 38     make: *** [makefile:267: lib/gl/renderer.o] Error 1
     39     make: *** Waiting for unfinished jobs....
  >> 40     make: *** [makefile:267: lib/gl/shader.o] Error 1
...
```


### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.22.0.dev0 (2064934ed3c02edeab05bbd5719df6acf5743784)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/14393772/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/14393773/spack-build-out.txt)

@goxberry @tomstitt @tzanio @v-dobrev

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/glvis/package.py
+++ b/var/spack/repos/builtin/packages/glvis/package.py
@@ -154,6 +154,7 @@ class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
             "CONFIG_MK={0}".format(self.spec["mfem"].package.config_mk),
         ]
 
+        # https://github.com/spack/spack/issues/42839
         result.append("CPPFLAGS=-DGLEW_NO_GLU")
 
         if self.spec.satisfies("@4.0:"):

--- a/var/spack/repos/builtin/packages/glvis/package.py
+++ b/var/spack/repos/builtin/packages/glvis/package.py
@@ -154,6 +154,8 @@ class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
             "CONFIG_MK={0}".format(self.spec["mfem"].package.config_mk),
         ]
 
+        result.append("CPPFLAGS=-DGLEW_NO_GLU")
+
         if self.spec.satisfies("@4.0:"):
             # Spack will inject the necessary include dirs and link paths via
             # its compiler wrapper, so we can skip them:

